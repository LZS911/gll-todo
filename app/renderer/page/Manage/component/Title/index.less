.title {
  .major-title {
    font-size: 18px;
  }
  .minor-title {
    margin-top: 10px;
    color: rgba(0, 0, 0, 0.3);
  }
  .link-title {
    cursor: pointer;
    font-size: 12px;
    margin-top: 10px;
    color: rgba(0, 0, 0, 0.3);
  }
  margin-bottom: 14px;
}
