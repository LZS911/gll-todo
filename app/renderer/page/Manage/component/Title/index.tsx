import React from 'react';
import { Space } from 'antd';
import './index.less';

interface ITitle {
  major: string;
  minor: string;
  link?: JSX.Element;
}
const Title: React.FC<ITitle> = props => {
  return (
    <div className="title">
      <div className="major-title">{props.major}</div>

      <Space>
        <div className="minor-title">{props.minor}</div>
        <div className="link-title">{props.link}</div>
      </Space>
    </div>
  );
};

export default Title;
