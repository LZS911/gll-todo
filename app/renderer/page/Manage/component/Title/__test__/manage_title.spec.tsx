import { render, screen } from '@testing-library/react';
import * as React from 'react';
import Title from '..';

describe('test title', () => {
  it('should render major and minor content', () => {
    const major = 'this is major';
    const minor = 'this is minor';

    const { container } = render(<Title major={major} minor={minor} />);

    screen.getByText(major);
    screen.getByText(minor);
    expect(container).toMatchSnapshot();
  });

  it('should render link content', () => {
    const major = 'this is major';
    const minor = 'this is minor';
    const link = <span>this is link</span>;

    const { container } = render(<Title major={major} minor={minor} link={link} />);

    screen.getByText(major);
    screen.getByText(minor);
    screen.getByText('this is link');
    expect(container).toMatchSnapshot();
  });
});
