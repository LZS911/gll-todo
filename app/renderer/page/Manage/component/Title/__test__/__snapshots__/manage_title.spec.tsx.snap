// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test title should render link content 1`] = `
<div>
  <div
    class="title"
  >
    <div
      class="major-title"
    >
      this is major
    </div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center"
    >
      <div
        class="ant-space-item"
        style="margin-right: 8px;"
      >
        <div
          class="minor-title"
        >
          this is minor
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          class="link-title"
        >
          <span>
            this is link
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test title should render major and minor content 1`] = `
<div>
  <div
    class="title"
  >
    <div
      class="major-title"
    >
      this is major
    </div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center"
    >
      <div
        class="ant-space-item"
        style="margin-right: 8px;"
      >
        <div
          class="minor-title"
        >
          this is minor
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          class="link-title"
        />
      </div>
    </div>
  </div>
</div>
`;
